File: 按钮组底层实现/buttonGroup.py
status = [0,0,0,0]
def change(idx):
    status[idx] = 1
    for index, item in enumerate(status):
        if index != idx:
            status[index] = 0
    return status
# 将第三个元素开启，同时其他元素关闭
print(change(3))

File: 按钮组底层实现/buttonGroup.lua
lights = {0,0,0,0};
function change(idx)
    lights[idx] = 1
	for key,value in pairs(lights) do
	    if(key ~= idx) then
			lights[key] = 0;
        end
	end
	return lights
end
change(3)
for key,value in pairs(lights) do
    print(value)
end